### PR TITLE
Use `super()` more in  backends

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

A similar PR to the others, wrt #3129. Not entirely sure all backends support this, so I guess we'll see how CI feels.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [n/a] New features are documented, with examples if plot related
- [n/a] Documentation is sphinx and numpydoc compliant
- [n/a] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [n/a] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_ps.py [hunk 1]
```diff
@@ -761,12 +761,10 @@ def _is_transparent(rgb_or_rgba):
 
 class GraphicsContextPS(GraphicsContextBase):
     def get_capstyle(self):
-        return {'butt': 0, 'round': 1, 'projecting': 2}[
-            GraphicsContextBase.get_capstyle(self)]
+        return {'butt': 0, 'round': 1, 'projecting': 2}[super().get_capstyle()]
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_ps.py:764]: Is there a reason not to
```suggestion
        return {'butt': 0, 'round': 1, 'projecting': 2}[self.get_capstyle()]
```

File: lib/matplotlib/backends/backend_ps.py [hunk 2]
```diff
@@ -761,12 +761,10 @@ def _is_transparent(rgb_or_rgba):
...
 
     def get_joinstyle(self):
-        return {'miter': 0, 'round': 1, 'bevel': 2}[
-            GraphicsContextBase.get_joinstyle(self)]
+        return {'miter': 0, 'round': 1, 'bevel': 2}[super().get_joinstyle()]
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_ps.py:767]:
```suggestion
        return {'miter': 0, 'round': 1, 'bevel': 2}[self.get_joinstyle()]
```
?

File: lib/matplotlib/backends/backend_ps.py (hunk 1 above)
Comment (QuLogic) [reply lib/matplotlib/backends/backend_ps.py:764]: Because this is `get_capstyle`, and that would be recursive.

Comment (timhoffm) [reply lib/matplotlib/backends/backend_ps.py:764]: Oh, :sheep:

**Pull request merged**